assets/core/annuaire_settings.vue

```diff
@@ -0,0 +1,482 @@
+                }
+            }
+        }
+    },
+    methods: {
+        send_credentials:function(){
+            this.credentials = !this.credentials;
+            axios.put("/annuaire/api/settings/1/",{ show_credentials: this.credentials },token)
+                .then(() => {
+                    this.$bvToast.toast("Sauvegardé.", {
+                        variant: "success",
+                        noCloseButton: true
+                    });
+                });
+        },
+        send: function(grp) {
+            const selected = this["selected_" + grp];
+            const groups_name = this["groups_name_" + grp];
+            const options = this["options_" + grp];
+            const groupOptions = this["groupOptions_" + grp];
+            for (let i in this.groups) {
+                if (this.groups[i].name == selected.name) {
+                    selected.id = this.groups[i].id;
+                }
+            }
+            for (let i in this.groups_name_responsibles) {
+                if (groups_name[i] == selected.name) {
+                    groups_name.splice(i, 1);
+                }
+            }
+            options.push(selected.name);
+            groupOptions.push(selected.id);
+            if (grp =="responsibles" || grp =="responsibles_sensitive"){
+                var adress = "can_see_" + grp;
+            } else {
+                var adress ="can_see_student_" + grp;
+            }
+            console.log(adress);
+            axios
+                .put(
+                    "/annuaire/api/settings/1/",
+                    { [adress]: groupOptions },
+                    token
+                ) //  /core/api/group/
+                .then(() => {
+                    selected.name = null;
+                    selected.id = null;
+                    this.$bvToast.toast("Sauvegardé.", {
+                        variant: "success",
+                        noCloseButton: true
+                    });
+                })
+                .catch(err => {
+                    alert(err);
+                });
+        },
+        deleteGroup: function(grp,item) {
+            var currentGroup = { id: null, name: null };
+            const groupOptions = this["groupOptions_" + grp];
+            const options = this["options_" + grp];
+            const grp_name = this["groups_name_" + grp];
+            this.$bvModal.msgBoxOk("Êtes-vous sûr de vouloir supprimer " + item +" ?")
+                .then(() => {
+                    for (let i in this.groups) {
+                        if (this.groups[i].name == item) {
+                            currentGroup.id = this.groups[i].id;
+                        }
+                    }
+                    for (let i in groupOptions) {
+                        if (currentGroup.id == groupOptions[i]) {
+                            groupOptions.splice(i, 1);
+                            options.splice(i, 1);
+                        }
+                    }
```

Comment:
Quand tu fais une recherche avec une boucle for si tu t'attend à avoir un seul résultat ou si seulement le premier est intéressant, alors lorsque que tu as trouvé l'élément en question, c'est un bon réflexe de mettre un break pour arrêter la boucle même si ce n'est pas strictement nécessaire.